### DISCUSSION TITLE
Python package streamlining

### DISCUSSION THREAD

**Discussion by achamayou** [microsoft/CCF#2909] [General]
https://pypi.org/project/ccf/ currently contains an assortment of components that originated from the test infrastructure and were found over time to be useful to a typical CCF operator, member or user.

As CCF has moved closer to standards in its user and governance interfaces, it is time to re-evaluate what belongs in this package. In a discussion with @eddyashton, @jumaffre and @wintersteiger, the following changes were proposed and received favourably:

1. ledger reading functionality is core to the ccf package, it will remain and continue to improve.
2. receipt validation isn't central, but is effectively going to become a dependency for 1. and so will stay as well.
3. client functionality is of dubious value at this point, now that CCF implements HTTP over TLS. Python users are better off using requests or httpx directly. The curl client is useful for testing purposes and will move back to the test infrastructure. Request signing code is useful sample, but is not representative of typical usage, and so does not belong in the official package either.
4. The proposal generator serves no real purpose at this point, now that proposals are JSON. It will be replaced by sample jinja templates.
5. All scripts (scurl, keygenerator etc) are useful samples, but are not representative of production usage of CCF today and so should not be part of the package either.

All items that are useful as samples will:

1. Move out to a different directory
2. Continue to be tested in CI
3. Be referenced in documentation if they aren't already

@lynshi @letmaik, and any other user with an interest in this matter, your input is very welcome.

**Comment by lynshi** [comment]
Sounds good to me, no complaints.
